.github/workflows/release-please.yml

```diff
@@ -4,45 +4,41 @@ on:
   push:
     branches:
       - main
-  pull_request:
+  workflow_dispatch:
```

Comment:
I see now why this was added. The `10.3.0` release was created (https://github.com/nodejs/node-gyp/releases/tag/v10.3.0) but never published. So this would allow us to publish from CI (and with `--provenance`) after these cases.

`node-gyp` currently does not have separate permissions for write access to the repo vs merging a release PR. If it did, then we would not want to add `workflow_dispatch` as a trigger because it would elevate anyone with write access to being able to publish a release.

Since we don't currently do this, I'm ok with adding this workflow_dispatch trigger since it is currently more important to be able to publish 😄

Reply:
I added this to allow triggered release-please manually. I think this doesn't expose publishing directly, as the `npm-publish` step still requires the release-please to be run first and actually "creates a releasee".

Reply:
That makes sense now and I don't think I was reading the workflow correctly.

> Taking a step back, I think this could be removed since the release-please is triggered on every push. So formally we should land PRs to re-trigger release.

Here is the issue I've seen in the past with release-please workflows:

1. The release workflow goes bad (for any reason) and is left in an inconsistent state. Currently this repo is in that state because we have a GitHub release for `10.3.0` but no npm publish for that versions
1. Make new commits to fix what went wrong with the release
2. `release-please` picks up those new commits and makes `10.3.1`
3. If it all works then `10.3.1` is correctly released and published everywhere

But this is not the ideal outcome because `10.3.0` was never published. So there needs to be some way to `npm publish`. That could be via `workflow_dispatch` or someone needs to do it manually.

Reply:
Actually, I think it could be "fine-ish" to just go ahead and publish `10.3.1` to make this easier 😮‍💨

Reply:
I agree (for various definitions of "fine-ish" 😄). In that case, I think `workflow_dispatch` won't be necessary to include in this PR?

But I do think it's an ok outcome to merge this and get CI fixed and have 10.3.1 published.

Since I have publish rights from my personal npm account, I would like to publish https://github.com/nodejs/node-gyp/commit/9afaf00e7ab547fd76c6890582abff9739effa24 from my local machine as `10.3.0` to keep a clean paper trail 🧹

Reply:
yeah, removed.